### ISSUE TITLE
W tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc:267] No versions of servable spellcheck found under base path /models/spellcheck

### ISSUE THREAD

**Issue by fzehracetin** [tensorflow/serving#1831]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux CentOS 8
- **TensorFlow Serving installed from (source or binary)**: with Docker
- **TensorFlow Serving version**: 1.15.0

I have a spellcheck model that I want to serve as a Rest service. My model running with TensorFlow version 1.15. In my windows machine, I used the same model directory to serve the model and it worked without a problem. Now, I want to serve from Centos 8 machine and it gives an error.

This is my command to run model in docker:
`
docker run -p 8501:8501 --mount type=bind,source=/serving/spellcheck,target=/models/spellcheck -e MODEL_NAME=spellcheck -t tensorflow/serving:1.15.0 &
`

I got this error:
`
W tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc:267] No versions of servable spellcheck found under base path /models/spellcheck
`

I also tried this command and I got the same error:
`
docker run -t --rm -p 8501:8501 -v "serving/spellcheck:/models/spellcheck" -e MODEL_NAME=spellcheck tensorflow/serving:1.15.0
`

Under serving/spellcheck directory this is the file structure:
```
--1
   -saved_model.pb
   -variables
      -variables.data-00000-of-00001  
      -variables.index
```

I installed tensorflow-serving with docker. This is the command:

`
docker pull tensorflow/serving:1.15.0
`

Full Trace:
```
2021-03-16 12:14:49.438859: I tensorflow_serving/model_servers/server.cc:85] Building single TensorFlow model file config:  model_name: spellcheck model_base_path: /models/spellcheck
2021-03-16 12:14:49.439173: I tensorflow_serving/model_servers/server_core.cc:462] Adding/updating models.
2021-03-16 12:14:49.439205: I tensorflow_serving/model_servers/server_core.cc:573]  (Re-)adding model: spellcheck
2021-03-16 12:14:49.439483: W tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc:267] No versions of servable spellcheck found under base path /models/spellcheck
2021-03-16 12:14:50.439716: W tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc:267] No versions of servable spellcheck found under base path /models/spellcheck

```

**Steps to Reproduce**
Docker runs a model in windows but gives an error in centos 8 machine.

**Attempted solution**

- Changing model directory to another place.
- Changing the docker run method with others.
- I already look at these sources and related issues: [Stackoverflow](https://stackoverflow.com/questions/45544928/tensorflow-serving-no-versions-of-servable-model-found-under-base-path), [Github](https://github.com/tensorflow/serving/issues/1419)

I really need help, thanks.

**Comment by fzehracetin** [comment]
When I try this command I got the same error.

`
docker run -t --rm -p 8501:8501 -v "/tfserving/tensorflow_serving/servables/tensorflow/testdata/saved_model_half_plus_two_cpu:/models/half_plus_two" -e MODEL_NAME=half_plus_two tensorflow/serving:1.15.0
`

```
2021-03-17 06:40:39.191778: I tensorflow_serving/model_servers/server.cc:85] Building single TensorFlow model file config:  model_name: half_plus_two model_base_path: /models/half_plus_two
2021-03-17 06:40:39.192062: I tensorflow_serving/model_servers/server_core.cc:462] Adding/updating models.
2021-03-17 06:40:39.192080: I tensorflow_serving/model_servers/server_core.cc:573]  (Re-)adding model: half_plus_two
2021-03-17 06:40:39.192893: W tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc:267] No versions of servable half_plus_two found under base path /models/half_plus_two
2021-03-17 06:40:40.193069: W tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc:267] No versions of servable half_plus_two found under base path /models/half_plus_two
```

I cloned tensorflow/serving repository with git clone:

`
git clone https://github.com/tensorflow/serving.git tfserving
`

After that, I switched to the 1.15 branch like this:

`
git checkout 748217e
`

**Comment by arghyaganguly** [comment]
@fzehracetin , were you able to install tf serving using docker in CentOS?
In #1591, it is mentioned that support for CentOS based docker images is not available.
Also [pr](https://github.com/tensorflow/serving/pull/156) is open regarding the same.

**Comment by fzehracetin** [comment]
Thank you for your reply. Yes, I installed with `docker pull tensorflow/serving:1.15.0` and it did not give any error. After installation, the container was running. 

Today, I am trying to build from the source. Do you think this will work with CentOS?

**Comment by arghyaganguly** [comment]
@fzehracetin , does the base path "/models/spellcheck" exist before  the docker run commands?

**Comment by fzehracetin** [comment]
@arghyaganguly No, '/models/spellcheck' path does not exist. My model is under source path "source=/serving/spellcheck"

But, if the problem was about the "/models/spellcheck" directory it should be able to serve the half_plus_two model provided by TensorFlow/serving. When I try to serve the Half_plus_two model it gave the same error as I mentioned before.

I have the latest version of Docker.

**Comment by fzehracetin** [comment]
I solved the problem. It was not about the docker. The problem is the model directory that I gave. When I define a variable for the directory the problem solved.

My solution: 

```
TESTDATA="$(pwd)/serving"
docker run -t --rm -p 8501:8501 -v "$TESTDATA/spellcheck:/models/spellcheck" -e MODEL_NAME=spellcheck tensorflow/serving:1.15.0 &
```

In Windows without defining dir variable tf serving works, but if you use Linux distributions defining variable is mandatory. This was actually a pretty basic problem but I searched the answer in different places.
